### DISCUSSION TITLE
Add Support for Custom FavIcon Images

### DISCUSSION THREAD

**Discussion by chernandezweb** [directus/directus#4184] [Feature Requests]
![image](https://user-images.githubusercontent.com/22950332/108524723-58317400-729d-11eb-80eb-318abc70d746.png)

The logo in the module bar gets updated, but the Favicon doesn't display anything.

I tried with a white png and svg 40x40, but still doesn't display the Favicon.

**Comment by benhaynes** [comment]
Converting this to a discussion since this is not actually a bug. :)

While it does make sense to display the PNG in the favicon, as of now we're only using the color in the favicon. When you add a logo we DO remove the Directus logo though.

The issue here is that a 40x40 image will likely not scale very well to 16x16. I think if we want to have custom favicons we should have a dedicated field for it within Project Settings.

**Reply by chernandezweb** [reply]
Oh Alright I see. 

In the doc it says that the logo is used to dynamically set as favicon so I tought it was a bug.

But yeah a dedicated field for the Favicon would be a nice feature :)

Thank you for the answer 👍

**Reply by rijkvanzanten** [reply]
(Updated the docs to reflect this 👍🏻  https://github.com/directus/directus/commit/35117389e56ba225b3b508316482293f635bfdd6)

**Reply by benhaynes** [reply]
Hmm, I'm not seeing this. The Docs reference the color for favicon.

https://docs.directus.io/guides/projects/#adjusting-project-settings

Is there somewhere else in the Docs that mentions the logo is used in the favicon?

**Reply by rijkvanzanten** [reply]
See the commit I linked :) 

<img width="1441" alt="CleanShot 2021-02-19 at 13 40 51@2x" src="https://user-images.githubusercontent.com/9141017/108547129-16fa8d80-72b8-11eb-82d6-ea5c7d64f311.png">

It used to say logo _and_ color, I changed it to just mention the color, which is accurate

**Comment by arladmin** [comment]
Having the option to upload an image for the favicon would be great!

**Comment by LasseRosenow** [comment]
Also I think it would be nice if you could also have a checkbox saying "use logo as favicon" instead of uploading 2 times the same icon.

Since svg favicons are a thing these days also normal logos can in fact scale very well down to be used as favicons aswell.

**Reply by rijkvanzanten** [reply]
> Since svg favicons are a thing these days also normal logos can in fact scale very well down to be used as favicons aswell.

Still crossing my fingers Webkit (Safari) will add support for SVG favicons someday.. 🤞🏻

**Reply by neo7-studio-web** [reply]
Or just a field to add a custom favicon for the browser

**Reply by benhaynes** [reply]
Not against this! It'd be nice to automate this, but more control is not a bad thing in this case.

**Reply by koksikus** [reply]
hey, I have a problem with favicon all the time. After clicking F5 - in second the Directus favicon appears, and then only a dot in the color of the project. Despite the fact that I have png 40x40 uploaded and set in the project. 
![Zrzut ekranu 2021-12-28 180014](https://user-images.githubusercontent.com/90528895/147599915-c97f82ab-e89e-45de-a08e-1cdb5a5be35e.png)

I do not know how to lead to another error maybe as I switch between roles for tests it happens that I am left with a Directus favicon. I have to click F5 to see a dot in the color of the project. Where to look for the problem?
![Zrzut ekranu 2021-12-28 201426](https://user-images.githubusercontent.com/90528895/147599952-c780e3bd-39e1-4b48-a34a-3546f665f0a1.png)

**Reply by rijkvanzanten** [reply]
The logo isn't used as the favicon currently. It uses your project color in the favicon (hence the color).

**Reply by koksikus** [reply]
Then why do I sometimes see a Directus favicon? Can you add a field in the settings where admin can enter a favicon file. Or admin can simply replace from the list of Directus files - this favicon with own - this is the simplest solution.

**Reply by rijkvanzanten** [reply]
That's the default. When a project logo is configured, it's replaced with a white-labelled alternative (the circle without logo). The configured project color is used as the color of the favicon.

> Or admin can simply replace from the list of Directus files - this favicon with own - this is the simplest solution.

This is actually not at simple solution at all.. The file is bundled in and served from the distributions in the node_modules folders. Every time you'd update the platform the files are replaced completely. It will have to come from a field in project settings 👍🏻

**Reply by koksikus** [reply]
Great idea with the favicon placed locally, just like other project files, and in the settings to be marked which one it is. I had experience with joomla before, and here it is done differently. Well, good luck in solving the problem.

**Reply by NilsBaumgartner1994** [reply]
This would be a very nice feature
+1

**Reply by dfanica** [reply]
I would have it the other way around. By default the logo uploaded should be used for favicon as well unless you tick the checkbox and upload another one. But, to be honest, the checkbox is just a nice to have and the logo being used as a favicon would be enough. It wouldn't even be that hard to do either since the directus logo is applied without a custom logo selected.

**Comment by AaronBeaudoin** [comment]
The colored circle in my web browser tab for my Directus project honestly almost looks like a glitch, like something is broken or not loading in properly. I know it isn't, but it'd be a lot more obvious if I could just specify an PNG image to use instead.

<img width="256" alt="image" src="https://user-images.githubusercontent.com/26592486/169355481-ae6e6656-b9a1-4d64-9f6c-572896f55225.png">

**Reply by benhaynes** [reply]
Happy to support this, but not an internal priority. Would love to discuss implementation and coordinate a PR.

**Reply by shaunchander** [reply]
Just to add onto this but most project colors would have a darker background to them considering it also acts as the background color for a custom logo.

On dark-themed browsers, the project color blends in with the tab bar making it visually hard to identify an open Directus tab.

For example, on my browser this is what I see: 

<img width="77" alt="Screen Shot 2022-06-11 at 1 23 15 PM" src="https://user-images.githubusercontent.com/5169985/173200327-c340ddd6-2c32-406e-b301-08c10ac008b6.png">

You could imagine, with more tabs, the more difficult it gets to visually identify your Directus tab.

**Reply by sxsws** [reply]
Thinking aloud and breaking this down: would this be
- add file upload field to project settings for favicon
- update main template (or all templates) to include the favicon?

**Reply by rijkvanzanten** [reply]
> Just to add onto this but most project colors would have a darker background to them considering it also acts as the background color for a custom logo.

I wouldn't want to make that assumption for all users

> Thinking aloud and breaking this down: would this be
> * add file upload field to project settings for favicon
> * update main template (or all templates) to include the favicon?

Yup! that's exactly it. The only other thing we'll have to add is the additional "allowed public key" to the /server/info endpoint, so we can read the favicon from the public role

**Comment by u12206050** [comment]
Just reviving this since today I found out that on safar the directus logo still shows even though I do have custom logo set.
![Screenshot 2022-09-30 at 15 24 51](https://user-images.githubusercontent.com/6641242/193279516-0893690a-4a9c-4a66-a7de-9f184b720ad3.png)

**Reply by sksar** [reply]
@rijkvanzanten @u12206050 
Shouldn't the `safari-pinned-tab.svg` and `mstile-150x150.png` files be removed as well;

This is with reference to #4184 
![Screenshot 2022-09-30 at 15 24 51](https://user-images.githubusercontent.com/6641242/193279516-0893690a-4a9c-4a66-a7de-9f184b720ad3.png)

The favicon does not work for safari and in some versions of Microsoft edge.
I believe we should remove them in favor of the normal favicon.ico.

And as a side note maybe a feature request to be allowed to upload a custom favicon, in the Directus settings.

_Originally posted by @sksar in https://github.com/directus/directus/issues/11952#issuecomment-1264220119_

**Reply by rijkvanzanten** [reply]
Yeah!

**Reply by sksar** [reply]
> Yeah!

Hmm. This is very random. It appears in some cases, and does not in some cases, even if it's the same computer and browser. Maybe it's best to remove it fully and have it user settable, or a fixed favicon, or no favicon.

**Reply by Cyon85** [reply]
Remove the functionality and use the static one, would be nice.
After that, it is possible to change it ... replace with an other file or a custom hook with the fav-route.

**Reply by benhaynes** [reply]
If done properly, I'm open to a PR that adds an override favicon image within the general settings. Unfortunately, this would be too low prior for us to take on internally in the short term.

**Comment by rijkvanzanten** [comment]
Heya! Thanks for opening this feature request!

This feature request has received over **15 votes** from the community. This means we'll move this feature request to the Under Review state! 

The Core team will schedule a meeting to review this request as soon as possible. The discussion will then be approved or denied. You may or may not be invited to join this meeting with the core team.

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._

**Comment by blackmotion55** [comment]
Yes, please.
I'm amazed to see how Directus is such an engineering masterpiece, yet details like that remain unchanged.
Thanks everyone

**Comment by w0kyj** [comment]
Add favicon to Project Settings -- Do as part of new theming update
https://github.com/directus/directus/issues/19916

**Comment by rijkvanzanten** [comment]
This was implemented and will ship as part of 10.7 very soon 😄

**Reply by benhaynes** [reply]
Yaassss!!!
